File: cpp/2.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *ret = new ListNode(-1), *pos = ret;
        int c = 0;
        while(l1||l2||c){
            int v = c;
            if(l1)v+=l1->val,l1=l1->next;
            if(l2)v+=l2->val,l2=l2->next;
            c=v/10;
            v%=10;
            pos->next = new ListNode(v);
            pos=pos->next;
        }
        return ret->next;
    }
};

File: js/2.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    var c = 0
    var ret = new ListNode(-1), pos = ret
    while(l1||l2||c) {
        var v = c
        if(l1)v+=l1.val,l1=l1.next
        if(l2)v+=l2.val,l2=l2.next
        if(v>=10){
            v-=10
            c=1
        }else c=0
        pos.next = new ListNode(v)
        pos = pos.next
    }
    return ret.next
};
